pkg/apis/monitoring/v1/podmonitor_types.go

```diff
@@ -131,6 +131,13 @@ type PodMonitorSpec struct {
 	//
 	// +optional
 	AttachMetadata *AttachMetadata `json:"attachMetadata,omitempty"`
+	// When defined, enforcedBodySizeLimit specifies a job level limit on the size
+	// of uncompressed response body that will be accepted by Prometheus.
+	//
+	// It requires Prometheus >= v2.28.0.
+	//
+	// +optional
+	EnforcedBodySizeLimit ByteSize `json:"enforcedBodySizeLimit,omitempty"`
```

Comment:
Since it's an optional field, we should use a pointer.
```suggestion
	EnforcedBodySizeLimit *ByteSize `json:"enforcedBodySizeLimit,omitempty"`
```

Reply:
Since `ByteSize` is a type alias of string and the zero value of string is "", should i still change it to `*ByteSize`?

Reply:
Given Simon's comment [here](https://github.com/prometheus-operator/prometheus-operator/pull/6303#discussion_r1492431876), you can skip it.